tests/unit/data_classes/test_api_gateway_proxy_event.py

```diff
@@ -0,0 +1,219 @@
+from aws_lambda_powertools.utilities.data_classes import (
+    APIGatewayProxyEvent,
+    APIGatewayProxyEventV2,
+)
+from tests.functional.utils import load_event
+
+
+def test_default_api_gateway_proxy_event():
+    event = APIGatewayProxyEvent(load_event("apiGatewayProxyEvent_noVersionAuth.json"))
```

Comment:
separate `raw_event` from `parsed_event`, otherwise you're testing Python getters of `APIGatewayProxyEvent` and not whether the data has been parsed correctly.

example

```python
raw_event = load_event("apiGatewayProxyEvent_noVersionAuth.json")
parsed_event = APIGatewayProxyEvent(**raw_event)

assert parsed_event.resource == raw_event["resource"]
```

Reply:
Hi @heitorlessa! This was lot of work, dude! hahaha